src/Bicep.LangServer/Handlers/ImportKubernetesManifestHandler.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.PrettyPrint.Options;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.Telemetry;
+using MediatR;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using SharpYaml.Serialization;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    [Method("bicep/importKubernetesManifest", Direction.ClientToServer)]
+    public record ImportKubernetesManifestRequest(string ManifestFilePath)
+        : IRequest<ImportKubernetesManifestResponse>;
+
+    public record ImportKubernetesManifestResponse(string? BicepFilePath);
+
+    public class ImportKubernetesManifestHandler : IJsonRpcRequestHandler<ImportKubernetesManifestRequest, ImportKubernetesManifestResponse>
+    {
+        private readonly TelemetryAndErrorHandlingHelper<ImportKubernetesManifestResponse> helper;
+
+        public ImportKubernetesManifestHandler(ILanguageServerFacade server, ITelemetryProvider telemetryProvider)
+        {
+            this.helper = new TelemetryAndErrorHandlingHelper<ImportKubernetesManifestResponse>(server.Window, telemetryProvider, new(null));
+        }
+
+        public Task<ImportKubernetesManifestResponse> Handle(ImportKubernetesManifestRequest request, CancellationToken cancellationToken)
+            => helper.ExecuteWithTelemetryAndErrorHandling(async () => {
+                var bicepFilePath = Path.ChangeExtension(request.ManifestFilePath, ".bicep");
+                var manifestContents = await File.ReadAllTextAsync(request.ManifestFilePath);
+
+                var bicepContents = Decompile(manifestContents);
+
+                await File.WriteAllTextAsync(bicepFilePath, bicepContents, cancellationToken);
+
+                return new(new(bicepFilePath), BicepTelemetryEvent.ImportKubernetesManifestSuccess());
+            });
+
+        public static string Decompile(string manifestContents)
+        {
+            manifestContents = StringUtils.ReplaceNewlines(manifestContents, "\n");
+
+            var resources = manifestContents.Split("\n---\n");
```

Comment:
There might be some edge cases here to handle. Most YAML parsers are able to handle the multi-document case. So if you can rely on the parser for this it will be better.

Reply:
Yeah, I felt bad for writing this. I couldn't make it work with the library but did not spend long figuring it out. Let me revisit that...